src/env.rs

```diff
@@ -0,0 +1,123 @@
+        let key = key.to_string();
+        match &self.mock_store {
+            Some(mock_store) => Ok(mock_store.get(&key).map(|v| v.to_owned())),
+            None => match env::var(&key) {
+                Ok(data) => Ok(Some(data)),
+                Err(e) => match e {
+                    env::VarError::NotPresent => Ok(None),
+                    env::VarError::NotUnicode(_) => Err(std::io::Error::new(
+                        std::io::ErrorKind::InvalidInput,
+                        format!(
+                            "The value of the environment variable \"{}\" is not valid Unicode.",
+                            &key
+                        ),
+                    )),
+                },
+            },
+        }
+    }
+
+    /// sets an environment variable to a value
+    pub fn set(&mut self, key: RoverEnvKey, value: &str) {
+        let key = key.to_string();
+        match &mut self.mock_store {
+            Some(mock_store) => {
+                mock_store.insert(key, value.into());
+            }
+            None => {
+                env::set_var(&key, value);
+            }
+        }
+    }
+
+    /// unsets an environment variable
+    pub fn unset_env(&mut self, key: RoverEnvKey) {
+        let key = key.to_string();
+        match &mut self.mock_store {
+            Some(mock_store) => {
+                mock_store.remove(&key);
+            }
+            None => {
+                env::remove_var(&key);
+            }
+        }
+    }
+}
+
+/// RoverEnvKey defines all of the environment variables
+/// that are respected by Rover. Any time a new environment variable
+/// is added to the public contract, it should be defined here.
+/// Each environment variable is prefixed with `APOLLO_` and
+/// the suffix is the name of the key defined here. It will automatically
+/// be converted from CamelCase to SHOUTY_SNAKE_CASE.
+/// For example, `RoverEnvKey::ConfigHome.to_string()` becomes `APOLLO_CONFIG_HOME`
+#[derive(Debug, Copy, Clone)]
+pub enum RoverEnvKey {
+    ConfigHome,
+    TelemetryUrl,
+    TelemetryDisabled,
+    Key,
+}
+
+impl fmt::Display for RoverEnvKey {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        let dbg = format!("{:?}", self).TO_SHOUTY_SNEK_CASE();
+        fmt.write_str(&format!("APOLLO_{}", &dbg))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn it_parses_config_home() {
+        let expected_key = "APOLLO_CONFIG_HOME";
+        assert_eq!(&RoverEnvKey::ConfigHome.to_string(), expected_key);
+    }
+
+    #[test]
+    fn it_can_set_and_read_from_mock() {
+        let expected_value = "hey whats the big idea anyway!??";
+        let key = RoverEnvKey::ConfigHome;
+        let mut env_store = RoverEnv::new();
+        env_store.set(key, expected_value);
+        let actual_value = env_store.get(key).unwrap().unwrap();
+        assert_eq!(expected_value, &actual_value)
+    }
+}
```

Comment:
this file is mostly what i'm looking for feedback on, the rest you can ignore for now